### PR TITLE
SPINEDEM-3046 Added logic to remove NHSD-Session-URID header if unneeded

### PR DESCRIPTION
**Pull request by PeterGallagherBJSS**
- Fixes an issue where this data was getting into SPINE SAR logs for requests where the header should be ignored.
- Minor doc changes.
- Removes the github action to add a comment with a link to the spec on the PR. This has been deprecated for some time.
- Dependabot dependencies.
- Fixes some failing tests that were written incorrectly.

## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.

### CODE REVIEW

**Review by ehsanhussain-nhsd — APPROVED** [review]

**Review by ehsanhussain-nhsd — DISMISSED** [review]

**Review by ehsanhussain-nhsd — APPROVED** [review]

**Review by ehsanhussain-nhsd — APPROVED** [review]

File: specification/personal-demographics.yaml [hunk 1]
```diff
@@ -928,7 +928,7 @@ paths:
 
         * the `id` - the object ID for items that have one - for example `name`, `address` or `telecom`.
         * the `url` - the URL for the extension being removed.
-        * the whole object being removed.
+        * the whole object being removed. (this is the only allowed option for patient access updates)
```
Comment (claudiudumitrascu1-nhs) [inline specification/personal-demographics.yaml:931]: @PeterGallagherBJSS , have this documentation changes been captured in the Jira ticket? We should get the wording reviewed by Jack

File: .github/workflows/pr-lint.yaml [hunk 2]
```diff
@@ -45,12 +45,3 @@ jobs:
           msg: |
             This branch is working on a ticket in the NHS Digital APM JIRA Project. Here's a handy link to the ticket:
             # [${{ env.TICKET_NAME }}](https://nhsd-jira.digital.nhs.uk/browse/${{ env.TICKET_NAME}})
-
-      - name: Comment on PR with link to Spec
```
Comment (claudiudumitrascu1-nhs) [inline .github/workflows/pr-lint.yaml:49]: Is the APIM producer support team not working on a fix for this?

File: specification/personal-demographics.yaml (hunk 1 above)
Comment (PeterGallagherBJSS) [reply specification/personal-demographics.yaml:931]: No I should've though. I'll ask him now.

Comment (PeterGallagherBJSS) [reply specification/personal-demographics.yaml:931]: Resolving as Jack is happy with the doc change

**Review by claudiudumitrascu1-nhs — COMMENTED** [review]
The current test strategy for personal-demographics-service is to test the proxy changes through pytests (anything else through karate tests), we should add such tests @PeterGallagherBJSS

**Pull request merged**